projects/store-test/src/app/app.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { actions } from '../store/user.store';
+
+@Component({
+	selector: 'app-root',
+	templateUrl: './app.component.html',
+	styleUrls: ['./app.component.scss'],
+})
+export class AppComponent {
+	title = 'store-test';
+
+	constructor(private readonly store: Store) {}
+
+	ngOnInit() {
+		this.store.dispatch(actions.users.effects.set());
```

Comment:
Misschien irrelevant, maar ik vroeg me af waarom het `.effect` level er hier tussen zit? :-) Dus waarom niet: 
```
this.store.dispatch(actions.users.set());
```

De effects "hake in" op actions die ge-emit worden, het lijkt me niet nodig om dat "zichtbaar" te maken in de actions. 🤔 Maar nogmaals, misschien niet relevant, of misschien begrijp het niet juist. :-)

Reply:
Simpelweg omdat set, add, update, delete vandaag al als losstaande actions bestaan en die dus al gebruikt worden. 

Ik moest dus een onderscheid maken tussen iets waarmee je het gewoon kan setten en een set kan triggeren dat via een effect verloopt. Ik twijfelde tussen die actions onder een aparte effects te steken of dan in de plaats ze prefixen met `trigger`.

Dus dan zou je dit krijgen:
```
this.store.dispatch(actions.users.triggerSet());
```

Ik heb geen voorkeur tbh, dus wat jullie het makkelijkste vind kan ik voorzien!

Reply:
Okay... In dat geval zou ik het wel anders doen... 😅 

In principe zou eender welke actie een effect moeten kunnen triggeren, het effect kan op zijn beurt dan opnieuw een actie triggeren, die vervolgens weer een ander effect triggert, afhankelijk van de use-case. Het feit dat de action een effect triggert, of zelf meteen via de reducer een stuk state veranderd is niet belangrijk voor het effect, ik zou dat onderscheid dus ook niet maken. 🙂 

Wij werken heel vaak met volgende acties per entity: INIT, SUCCES, ERROR, CLEAR

Meestal gaat bij ons enkel de `INIT` actie een effect triggeren, bijvoorbeeld een HTTP call, simultaan wordt er ook in de reducer geluisterd op de `INIT` actie en wordt de `loading` property op true gezet en de `result` prop (kan ook `list`, of `detail` of ... zijn), wordt op `null` of `[]` gezet. Vervolgens is de HTTP call uitgevoerd, die kan succesvol zijn of falen en o.b.v. die state zal het effect de `ERROR` of `SUCCESS` action dispatchen die vervolgens opgepikt wordt in de reducer en de state op de juiste manier aanpast. Wij maken dus nergens een referentie vanuit de actions naar de effects. 🙂

Reply:
Ik snap dat jullie die manier van werking hebben; maar ik moet dat onderscheid wel maken binnen de package, daar kan ik niet om heen 😅 . Zowel e-loket en v-loket werken zonder effects, dus het moet blijven werken. En ergens vind ik ook dat we de keuze om effects niet moeten opdringen bij anderen, dus het moet volledig zonder effects blijven werken gezien die uiteindelijk op zich nog altijd optioneel zijn binnen NGRX.

Wat ik kan doen is dat je specifieke assets kan genereren die _enkel_ effects based actions maakt; zodat je bv `generateBaseStoreAssetsWithEffects` kan doen. Maar daar ben ik dan geen fan van, want dan heb je een property in je store die _enkel_ met effects kunnen werken. 

Ik snap dat jullie het patroon gebruiken die je al hebt in EBMT; en dit is natuurlijk een lichtjes andere manier van implementatie. Maar in de praktijk zou het wel echt exact hetzelfde moeten kunnen als wat je vandaag hebt. 

Misschien moeten we dit eens overlopen in een call :D